sql/wsrep_schema.cc

```diff
@@ -1535,3 +1537,54 @@ int Wsrep_schema::recover_sr_transactions(THD *orig_thd)
 out:
   DBUG_RETURN(ret);
 }
+
+int Wsrep_schema::store_gtid_event(THD* thd,
+                                   const Gtid_log_event *gtid)
+{
+  int error=0;
+  void *hton= NULL;
+  const bool in_transaction= (gtid->flags2 & Gtid_log_event::FL_TRANSACTIONAL);
+  const bool in_ddl= (gtid->flags2 & Gtid_log_event::FL_DDL);
+  Wsrep_schema_impl::wsrep_off  wsrep_off(thd);
```

Comment:
Turning wsrep off here causes the first statement to run with wsrep OFF. If `mysql.gtid_slave_pos` table is InnoDB table, this causes the transaction to be started with wsrep OFF. As the wsrep status is synced only at the beginning of the transaction in InnoDB, this causes the rest of the transaction to run with wsrep OFF too.